Author: ANDK (Andreas Koenig)
Description: DateTime used to silently accept non-integer arguments and not do anything
 intelligent with them. Since 0.52 it insists on integers.
Origin: ftp://cpan.cpantesters.org/CPAN/authors/id/A/AN/ANDK/patches/DateTime-Format-DateParse-0.04-ANDK-01.patch.gz
Bugs: http://rt.cpan.org/Public/Bug/Display.html?id=52598
Bugs-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=560637

--- a/lib/DateTime/Format/DateParse.pm
+++ b/lib/DateTime/Format/DateParse.pm
@@ -30,9 +30,10 @@
 
     my %p;
     if ( $ss ) {
-        my $fraction = $ss - int( $ss );
-        $p{ nanosecond } = $fraction * 1e9 if $fraction;
-        $p{ second } = int $ss;
+        ($p{"second"},my $nanosecond) = split /\./, $ss;
+        if ( $nanosecond ){
+            $p{ nanosecond } = $nanosecond . "0"x(9-length($nanosecond));
+        }
     }
     $p{ minute }    = $mm if $mm;
     $p{ hour }      = $hh if $hh;
